dev/tests/js/jasmine/tests/app/code/Magento/Customer/frontend/js/customer-data.test.js

```diff
@@ -95,84 +97,100 @@ define([
     }
 
     describe('Magento_Customer/js/customer-data', function () {
+
         beforeAll(function () {
             clearLocalStorage();
         });
 
-        beforeEach(function () {
-            jQueryGetJSON = $.getJSON;
+        beforeEach(function (done) {
+            originalGetJSON = jQuery.getJSON;
             sectionConfig['Magento_Customer/js/section-config'](sectionConfigSettings);
+
+            injector.require([
+                'Magento_Customer/js/customer-data'
+            ], function (Constr) {
+                obj = Constr;
+                done();
+            });
```

Comment:
Seems like we don't really need to mock any dependencies, so looks like we can just use customerData object that already imported above. What do you thing?

Reply:
Removing the mocking breaks the following:

`Check it requests sections from the server if there are expired sections` with error:
```
Expected spy $.getJSON to have been called with [ 'http://localhost/customer/section/load/', <jasmine.objectContaining(Object({ sections: 'cart,customer' }))> ] but it was never called. (1)
```

`"getExpiredSectionNames method` -> `Does not throw before component is initialized` with error:
```
Expected function not to throw, but it threw TypeError: null is not an object (evaluating 'i[a[f]]'). (1)
```

`"set" method` -> `Does not throw before component is initialized...`
```
>> Error: [object Object] at
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/Magento_Customer/js/customer-data.js:87
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:3238 fire
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:3368 fireWith
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:9848 done
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:10317 callback
>> Error: [object Object] at
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/Magento_Customer/js/customer-data.js:87
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:3238 fire
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:3368 fireWith
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:9848 done
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:10317 callback
```

Reply:
Nope it was needed, using just customerData didn't end up working, I also ended up needing to inject underscore here to resolve another one of your suggestions below.